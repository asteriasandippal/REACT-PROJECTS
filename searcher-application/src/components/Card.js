import React from 'react';

function Card(props) {
    return (
        <div className="card">
            <img className="card-img-top" src={props.image_url} alt={props.name}/>
            <div className="card-body">
                {props.name ? <h5 className="card-title">{props.name}</h5> : ''}
                {props.description ? <p className="card-text">{props.description}</p> : ''}
            </div>
        </div>        
    );
}

export default Card;
