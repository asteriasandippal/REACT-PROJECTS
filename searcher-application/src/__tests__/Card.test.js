import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';


import Card from '../components/Card';

test('Card renders an img tag with the correct src', () => {
    const component = renderer.create(
        <Card 
            image='http://test.com/test.jpg' 
            name='Card Title'
            description='Card Description'
        />
    );

    const tree = component.toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
    expect(tree.props.className).toBe('card');
    // expect(tree.props.img).toBe('http://test.com/test.jpg');
    // expect(tree.props.name).toBe('Card Title');
    // expect(tree.props.description).toBe('Card Description');
    
});
