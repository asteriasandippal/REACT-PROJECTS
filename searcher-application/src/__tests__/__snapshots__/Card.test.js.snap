// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders an img tag with the correct src 1`] = `
<div
  className="card"
>
  <img
    alt="Card Title"
    className="card-img-top"
    src="http://test.com/test.jpg"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Card Title
    </h5>
    <p
      className="card-text"
    >
      Card Description
    </p>
  </div>
</div>
`;
